arch/arm/soc/nxp_kinetis/kwx/Makefile

```diff
@@ -5,4 +5,7 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-obj-y += soc.o
+obj-$(CONFIG_SOC_MKW24D5) += wdog.o soc_kw2xd.o
+obj-$(CONFIG_SOC_MKW22D5) += wdog.o soc_kw2xd.o
+obj-$(CONFIG_SOC_MKW41Z4) += soc_kw4xz.o
+obj-$(CONFIG_SOC_MKW40Z4) += soc_kw4xz.o
```

Comment:
I think eventually we should try to unify these into one soc.c file, but it's fine to have them separate for now. I'm planning to work on a more general MCG clock driver soon that should help.